src/convertToObject.js

```diff
@@ -6,7 +6,23 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString.length === 0) {
+    return {};
+  }
+
+  const properties = sourceString.split(';');
+
+  return properties
+    .map((property) => property.trim())
+    .filter((property) => property.length > 0)
+    .reduce((prev, property) => {
+      const [key, value] = property.split(':');
+
+      return {
+        ...prev,
+        [key.trim()]: value.trim(),
```

Comment:
[CODE KNOWLEDGE] - Don't use iteration methods to modify an already existing array/object. In this case, the spread operator inside the reduce callback is creating a new object on each iteration, which is not efficient. Consider using the `Object.assign` method or directly assigning the new key-value pair to the `prev` object.